crates/valence_core/src/game_mode.rs

```diff
@@ -10,3 +10,15 @@ pub enum GameMode {
     Adventure,
     Spectator,
 }
+
+impl GameMode {
+    /// Converts gamemode to its number representation
+    /// ### Values
+    /// 0. Survival
+    /// 1. Creative
+    /// 2. Adventure
+    /// 3. Spectator
+    pub fn to_index(self) -> usize {
+        self as usize
+    }
```

Comment:
It will be used in `EntitySelector` because I want to store `gamemode` field as an array `[bool; 4]` and `GameMode::to_index` is the index. (I was already doing `EntitySelector` but deleted this code because It was done with another `Parse` trait)

Reply:
I dislike this because it makes the order of values in `GameMode` matter, and `EntitySelector` would have to be tweaked if a new game mode is added.

Reply:
The order of variants already matters because `GameMode` is `#[derive(Encode, Decode)]`. If you must keep this `to_index` function I would prefer it moved to `valence_command`.